seminars/seminar_04/sem_04_07.py

```diff
@@ -4,9 +4,19 @@
 # ✔ Вычислите итоговую прибыль или убыток каждой компании. Если все компании
 # прибыльные, верните истину, а если хотя бы одна убыточная — ложь.
 
-# TODO использовать all()
-
-
+def calc_profit(companies: dict) -> bool:
+    '''
+    Определение прибыльности/убыточности компаний.
+    :param companies: dict(name of company: [financial result)]
+    :return: bool (True - все компании прибыльные или сработали в ноль, False - хотя бы одна убыточная)
+    '''
+    return all([sum(value) >= 0 for value in companies.values()])
```

Comment:
Отличное решение, но `list comprehention` не нужен. Достаточно простого генератора
```suggestion
    return all(sum(value) >= 0 for value in companies.values())
```
а еще тут отлично подойдет функция map. Попробуйте